src/neo/SmartContract/Native/Votes/Method/FindAllPathMethod.cs

```diff
@@ -0,0 +1,207 @@
+        /// <returns>path set</returns>
+        public List<List<int>> FindPath(int startNode, int endNode)
+        {
+            path.Clear();
+            List<int> tempUnUsedNodes = new List<int>();
+            List<int> tempPath = new List<int>();
+            tempUnUsedNodes.AddRange(nodesList);
+            tempUnUsedNodes.Remove(startNode);
+            CreatePath(tempUnUsedNodes, startNode, endNode, tempPath);
+            return path;
+        }
+
+        /// <summary>
+        /// Create all paths from current node to end node
+        /// </summary>
+        /// <param name="unUsedNodes">unused node set</param>
+        /// <param name="currentNode">current node</param>
+        /// <param name="endNode">end node</param>
+        /// <param name="hasCreatedPath">the path has been created</param>
+        private void CreatePath(List<int> unUsedNodes, int currentNode, int endNode,
+                                      List<int> hasCreatedPath)
+        {
+            if (currentNode.Equals(endNode))
+            {
+                List<int> tempPath = new List<int>();
+                tempPath.AddRange(hasCreatedPath);
+                tempPath.Add(currentNode);
+                path.Add(tempPath);
+                return;
+            }
+            //calcualte the next node can be arrived
+            int startNodeIndex = System.Convert.ToInt32(currentNode);
+            List<int> achievedNodes = new List<int>();
+            foreach (int unUsedNode in unUsedNodes)
+            {
+                int achievedNodeIndex = System.Convert.ToInt32(unUsedNode);
+                if (nodesEdgesMatrix[startNodeIndex, achievedNodeIndex] > 0)
+                {
+                    achievedNodes.Add(unUsedNode);
+                }
+            }
+            //Exit condition, reaching the end node
+            if (achievedNodes.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                //Continue to traverse the node
+                foreach (int achievedNode in achievedNodes)
+                {
+                    List<int> tempUnUsedNodes = new List<int>();
+                    tempUnUsedNodes.AddRange(unUsedNodes);
+                    tempUnUsedNodes.Remove(achievedNode);
+                    List<int> tempPath = new List<int>();
+                    tempPath.AddRange(hasCreatedPath);
+                    tempPath.Add(currentNode);
+                    CreatePath(tempUnUsedNodes, achievedNode, endNode, tempPath);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Find all paths in  vector graphic
+        /// </summary>
+        /// <returns>path set</returns>
+        public List<List<int>> FindAllPath()
+        {
+            path.Clear();
+            foreach (int i in nodesList)
+            {
+                foreach (int j in nodesList)
+                {
+                    if (!i.Equals(j))
```

Comment:
I think this is not the best way to traverse all paths of a directed graph. Plz use dfs.
Is it possible to combine `CreatPArray` into this dfs.